os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -0,0 +1,985 @@
+ ****************************************************************************/
+/* USB trace dumping */
+
+#ifndef CONFIG_USBDEV_TRACE
+#undef CONFIG_ARCH_USBDUMP
+#endif
+
+#define FRAME_POINTER_ADDR __builtin_frame_address(0)
+
+/* Flag used to detect the whether upassert called or not */
+bool g_upassert = false;
+
+/****************************************************************************
+ * Public Variables
+ ****************************************************************************/
+char assert_info_str[CONFIG_STDIO_BUFFER_SIZE] = {'\0', };
+
+/****************************************************************************
+ * Private Data
+ ****************************************************************************/
+
+/****************************************************************************
+ * Private Functions
+ ****************************************************************************/
+
+/****************************************************************************
+ * Name: up_stackdump
+ ****************************************************************************/
+
+#ifdef CONFIG_ARCH_STACKDUMP
+static void up_stackdump(uint32_t sp, uint32_t stack_base)
+{
+	//PORTNOTE :- Most/all functions in this file are dummy rn
+	//Dummy for now
+	//Needs to be implemented according to architecture
+	return;
+
+	uint32_t stack = sp & ~0x1f;
+	uint32_t *ptr = (uint32_t *)stack;
+	uint8_t i;
+
+	lldbg("%08x:", stack);
+	for (i = 0; i < 8; i++) {  // Print first 8 values for sp located
+		if (stack < sp) {
+			/* If stack pointer(sp) is aligned(sp & 0x1f) to an address outside allocated stack */
+			/* Then, for stack addresses outside allocated stack, print 'xxxxxxxx' */
+			lldbg_noarg(" xxxxxxxx");
+		} else {
+			/* For remaining stack addresses inside allocated stack, print proper stack address values */
+			lldbg_noarg(" %08x", ptr[i]);
+		}
+		stack += 4;
+	}
+	lldbg_noarg("\n");
+
+	for (; stack < stack_base; stack += 32) { // Print remaining stack values from 9th value to end
+		ptr = (uint32_t *)stack;
+		lldbg("%08x: %08x %08x %08x %08x %08x %08x %08x %08x\n",
+			   stack, ptr[0], ptr[1], ptr[2], ptr[3], ptr[4], ptr[5], ptr[6], ptr[7]);
+	}
+}
+#else
+#define up_stackdump(sp, stack_base)
+#endif
+
+typedef struct regs {
+	uint32_t reg[ARM_CONTEXT_REGS];	/* For 32 bit processor, R0 to R15 and CPSR total 16 */
+} arm_regs_t;
+
+arm_regs_t task_ctxt_regs;		/* Global variable to store the register context */
+struct arm_mode_regs_s cpu_ctxt_regs;	/* Global variable to store the register context */
+
+#ifdef CONFIG_FRAME_POINTER
+struct stackframe {
+	uint32_t programCounter;	// R15
+	uint32_t linkRegister;		// R14
+	uint32_t stackPointer;		// R13
+	uint32_t framePointer;		// R11
+};
+
+/****************************************************************************
+ * Name: is_text_address
+ ****************************************************************************/
+
+static int is_text_address(unsigned long programCounter)
```

Comment:
Remove this API please.
Kindly check latest assert code in v8-m and update accordingly.